en/lessons/ecto/querying.md

```diff
@@ -90,114 +90,89 @@ iex> Repo.all(query)
+The very important thing here is how output of the query changed. Using an *expression* with a binding in `select:` part allows you to specify exactly the way selected fields will be returned. We can ask for a tuple, for example:
 
 ```elixir
-iex(15)> query = from(m in Movie, select: m.title)
-%Ecto.Query<from m in Example.Movie, select: m.title>
-iex(16)> Repo.all(query)
+iex> query = from(m in Movie, where: m.id < 2, select: {m.title})             
 
-15:06:12.752 [debug] QUERY OK source="movies" db=4.5ms queue=0.1ms
-["Ready Player One"]
+iex> Repo.all(query)                                                          
+[{"Ready Player One"}]
 ```
 
-Notice that this approach to using `select` returns a list containing the selected values.
+It is a good idea to always start with a simple bindingless query and introduce a binding whenever you need to reference your data structure. More on bindings in queries can be found in [Ecto documentation](https://hexdocs.pm/ecto/Ecto.Query.html#module-query-expressions)
 
-### Using `where` expressions
 
-We can use `where` expressions to include "where" clauses in our queries. Multiple `where` expressions are combined into `WHERE AND` SQL statements.
+### Macro-based queries
 
-```elixir
-iex> query = from(m in Movie, where: m.title == "Ready Player One")
-%Ecto.Query<from m in Example.Movie, where: m.title == "Ready Player One">
-iex> Repo.all(query)
+In the examples above we used keywords `select:` and `where:` inside of `from` macro to build a query – these are so called *keyword-based queries*. There is, however, another way to compose queries – macro-based queries. Ecto provides macros for every keyword, like `select/3` or `where/3`. Each macro accepts a *queryable* value, *an explicit list of bindings* and the same expression you'd provide to its keyword analogue:
 
-15:18:35.355 [debug] QUERY OK source="movies" db=4.1ms queue=0.1ms
-[
-  %Example.Movie{
-    __meta__: %Ecto.Schema.Metadata<:loaded, "movies">,
-    actors: %Ecto.Association.NotLoaded<association :actors is not loaded>,
-    characters: %Ecto.Association.NotLoaded<association :characters is not loaded>,
-    distributor: %Ecto.Association.NotLoaded<association :distributor is not loaded>,
-    id: 1,
-    tagline: "Something about video games",
-    title: "Ready Player One"
-  }
-]
+```elixir
+iex> query = select(Movie, [m], m.title)                           
+%Ecto.Query<from m in Example.Movie, select: m.title>
+iex> Repo.all(query)                    
+SELECT m0."title" FROM "movies" AS m0 []
+["Ready Player One"]
 ```
 
-We can use `where` expressions together with `select`:
+The good thing about macros is that they work very well with pipes:
 
 ```elixir
-iex> query = from(m in Movie, where: m.title == "Ready Player One", select: m.tagline)
-%Ecto.Query<from m in Example.Movie, where: m.title == "Ready Player One", select: m.tagline>
-iex> Repo.all(query)
+query = Movie
```

Comment:
Maybe, someone got `(SyntaxError) iex:2: syntax error before: '|>'`error here.
I think `iex> query = Movie |> where([m], m.id < 2) |> select([m], {m.title})` will be more kind.